### DISCUSSION TITLE
How to pass template strings into sql query?

### DISCUSSION THREAD

**Discussion by stalinraj237** [prisma/prisma#23415] [Q&A] [answered]
### Question

I have the below array of numbers:
```js
let levels = [86, 88, 89, 90];
let levelsConverted:any = '(' + levels.join(', ') + ')';
```

Now I need to pass levelsConverted values to my sql insert query.
```js
return await this.prisma.$queryRaw`
            `DECLARE @levelList TABLE (id int);`
           ` INSERT INTO @levelList(id) VALUES ${levelsConverted};`
            `SELECT * FROM Users WHERE LevelId IN (SELECT id FROM @levelList)``;`
```
when I run the code am getting the below error:
**PrismaClientKnownRequestError:
Invalid `prisma.$queryRaw()` invocation:
Raw query failed. Code: `102`. Message: `Incorrect syntax near '@P1'.`**

Need some valuable help.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @stalinraj237 👋 

Can you please try this modified query?

```js
let levels = [86, 88, 89, 90];
let levelsConverted = levels.map(level => `(${level})`).join(', ');

return await this.prisma.$queryRaw`
            DECLARE @levelList TABLE (id int);
            INSERT INTO @levelList(id) VALUES ${Prisma.raw(levelsConverted)};
            SELECT * FROM Users WHERE LevelId IN (SELECT id FROM @levelList);`
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nikolasburk** [reply]
Hey @stalinraj237 👋 just a little tip: If you find yourself using `$queryRaw` a lot in your code, you may want to check out this Prisma Client extension that lets you build SQL queries for Prisma Client in a type-safe way 😄 

https://github.com/eoin-obrien/prisma-extension-kysely
